linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -354,51 +356,116 @@ object Infos {
+    def addMethodCalledStatically(method: NamespacedMethodName): this.type = {
+      methodsCalledStatically += method
+      this
+    }
 
-      new ReachabilityInfo(
-          fieldsRead = toMapOfLists(fieldsRead),
-          fieldsWritten = toMapOfLists(fieldsWritten),
-          staticFieldsRead = toMapOfLists(staticFieldsRead),
-          staticFieldsWritten = toMapOfLists(staticFieldsWritten),
-          methodsCalled = toMapOfLists(methodsCalled),
-          methodsCalledStatically = toMapOfLists(methodsCalledStatically),
-          methodsCalledDynamicImport = toMapOfLists(methodsCalledDynamicImport),
-          jsNativeMembersUsed = toMapOfLists(jsNativeMembersUsed),
-          instantiatedClasses = instantiatedClasses.toList,
-          accessedModules = accessedModules.toList,
-          usedInstanceTests = usedInstanceTests.toList,
-          accessedClassData = accessedClassData.toList,
-          referencedClasses = referencedClasses.toList,
-          staticallyReferencedClasses = staticallyReferencedClasses.toList,
-          accessedClassClass = accessedClassClass,
-          accessedNewTarget = accessedNewTarget,
-          accessedImportMeta = accessedImportMeta,
-          usedExponentOperator = usedExponentOperator
+    def addMethodCalledDynamicImport(method: NamespacedMethodName): this.type = {
+      methodsCalledDynamicImport += method
+      this
+    }
+
+    def addJSNativeMemberUsed(member: MethodName): this.type = {
+      jsNativeMembersUsed += member
+      this
+    }
+
+    private def setFlag(flag: ReachabilityInfoInClass.Flags): this.type = {
+      flags |= flag
+      this
+    }
+
+    def setInstantiated(): this.type =
+      setFlag(ReachabilityInfoInClass.FlagInstantiated)
+
+    def setModuleAccessed(): this.type =
+      setFlag(ReachabilityInfoInClass.FlagModuleAccessed)
+
+    def setInstanceTestsUsed(): this.type =
+      setFlag(ReachabilityInfoInClass.FlagInstanceTestsUsed)
+
+    def setClassDataAccessed(): this.type =
+      setFlag(ReachabilityInfoInClass.FlagClassDataAccessed)
+
+    def setStaticallyReferenced(): this.type =
+      setFlag(ReachabilityInfoInClass.FlagStaticallyReferenced)
+
+    def result(): ReachabilityInfoInClass = {
+      new ReachabilityInfoInClass(
+          className,
+          fieldsRead = toLikelyEmptyList(fieldsRead),
+          fieldsWritten = toLikelyEmptyList(fieldsWritten),
+          staticFieldsRead = toLikelyEmptyList(staticFieldsRead),
+          staticFieldsWritten = toLikelyEmptyList(staticFieldsWritten),
+          methodsCalled = toLikelyEmptyList(methodsCalled),
+          methodsCalledStatically = toLikelyEmptyList(methodsCalledStatically),
+          methodsCalledDynamicImport = toLikelyEmptyList(methodsCalledDynamicImport),
+          jsNativeMembersUsed = toLikelyEmptyList(jsNativeMembersUsed),
+          flags = flags
       )
     }
+
+    private def toLikelyEmptyList[A](set: mutable.Set[A]): List[A] =
```

Comment:
Does this change the execution time? At least for 2.13, the `toList` call seems to end up here (with `Nil` as receiver):

https://github.com/scala/scala/blob/abb224a35fa7c386671eb7fb941e870bde0ec447/src/library/scala/collection/immutable/List.scala#L146-L165

Reply:
In 2.12 that goes to `TraversableOnce.toList`, which then goes to `TraversableLike.to[List]`:
https://github.com/scala/scala/blob/e66a4617c980314dd1c47c68077c7f8ae2e4183e/src/library/scala/collection/TraversableLike.scala#L782-L788
which is not great. It invokes a `CanBuildFrom`, creates a builder, and sends itself to it with `++= this`.

I can't really test whether it changes the execution time right now. I'm on Windows during the week-end, and timings are a lot less reliable.

Reply:
Ah, yeah, fair enough. Then let's leave it.